src/coreclr-debug/debugConfigurationProvider.ts

```diff
@@ -59,7 +63,64 @@ export class DotnetDebugConfigurationProvider implements vscode.DebugConfigurati
                 return undefined;
             }
         }
+        
+        // We want to ask the user if we should run dotnet  dev-certs https --trust, but this doesn't work in a few cases --
+        // Linux -- not supported by the .NET CLI as there isn't a single root cert store
+        // VS Code remoting/Web UI -- the trusted cert work would need to happen on the client machine, but we don't have a way to run code there currently
+        // pipeTransport -- the dev cert on the server will be different from the client
+        if (!this.platformInformation.isLinux() && !vscode.env.remoteName && vscode.env.uiKind != vscode.UIKind.Web && !debugConfiguration.pipeTransport)
+        {
+            if(debugConfiguration.checkForDevCert === undefined && debugConfiguration.serverReadyAction)
+            {
+                debugConfiguration.checkForDevCert = true;
+            }
+
+            if (debugConfiguration.checkForDevCert)
+            {
+                checkForDevCerts(this.options.dotNetCliPaths, this.eventStream);
+            }
+        }
 
         return debugConfiguration;
     }
 }
+
+function checkForDevCerts(dotNetCliPaths: string[], eventStream: EventStream){
+    hasDotnetDevCertsHttps(dotNetCliPaths).then(async (returnData) => {
```

Comment:
>hasDotnetDevCertsHttps [](http://example.com/codeflow?start=4&length=22)

Will this throw if `dotnet` is not found?

Reply:
it won't throw as execChildProcess() will just include the error in the object it returns, but because we only check for the presence of returnData.error we will still show the "dev certificate not found" message. I think it should check for the exact error code (6), and ignore otherwise.